src/typechecker/derive-type.lisp

```diff
@@ -5,6 +5,8 @@
 ;;;
 
 (defun derive-literal-type (value)
+  "Derive the type of a given literal VALUE. 
+  Return two values, the type (a TY) and a list of predicates (a TY-PREDICATE-LIST)."
```

Comment:
Sorry to be so picky, but please don't break lines in this way. I would suggest either:

1. Not breaking lines unless you're starting a new paragraph, in which case you use two new lines with no indentation. This is my preferred style, but is an eye sore to many. 

or:

2. Break lines (but still respecting the aforementioned paragraph rule), using `M-q` in Emacs, autowrapping at 80ish columns. This is not my preferred style, but is easier to read in an editor. 

There basically should never be a newline and whitespace.